src/main/webapp/js/maps-page-loader.js

```diff
@@ -39,5 +44,85 @@ function initConfigMap() {
     // Associate the styled map with the MapTypeId and set it to display.
     map.mapTypes.set(MAP_NIGHT_TYPE_ID, styledMapType);
     map.setMapTypeId(MAP_NIGHT_TYPE_ID);
+
+    // When the user clicks in the map, show a marker with a text box the user can edit.
+    map.addListener('click', (event) => {
+      createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+    });
+    fetchMarkers();
+  });
+}
+
+/** Fetches markers from the backend and adds them to the map. */
+function fetchMarkers(){
+  fetch('/markers').then((response) => {
+    return response.json();
+  }).then((markers) => {
+    markers.forEach((marker) => {
+     createMarkerForDisplay(marker.lat, marker.lng, marker.content)
+    });
+  });
+}
+
+/** Creates a marker that shows a read-only info window when clicked. */
+function createMarkerForDisplay(lat, lng, content){
+  const marker = new google.maps.Marker({
+    position: {lat: lat, lng: lng},
+    map: map
   });
+  var infoWindow = new google.maps.InfoWindow({
+    content: content
+  });
+  marker.addListener('click', () => {
+    infoWindow.open(map, marker);
+  });
+}
+
+/** Sends a marker to the backend for saving. */
+function postMarker(lat, lng, content){
+  const params = new URLSearchParams();
+  params.append('lat', lat);
+  params.append('lng', lng);
+  params.append('content', content);
+  fetch('/markers', {
+    method: 'POST',
+    body: params
+  });
+}
+
+/** Creates a marker that shows a textbox the user can edit. */
+function createMarkerForEdit(lat, lng){
+  // If an editable marker is already shown, remove it.
+  if (editMarker) {
+   editMarker.setMap(null);
+  }
+  editMarker = new google.maps.Marker({
+    position: {lat: lat, lng: lng},
+    map: map
+  });
+  const infoWindow = new google.maps.InfoWindow({
+    content: buildInfoWindowInput(lat, lng)
+  });
+  // Removes the marker when the user closes the editable info window.
+  google.maps.event.addListener(infoWindow, 'closeclick', () => {
+    editMarker.setMap(null);
+  });
+  infoWindow.open(map, editMarker);
+}
+
+/** Builds and returns HTML elements that show an editable textbox and a submit button. */
+function buildInfoWindowInput(lat, lng){
+  const textBox = document.createElement('textarea');
+  const button = document.createElement('button');
+  button.appendChild(document.createTextNode('Submit'));
+  button.onclick = () => {
+    postMarker(lat, lng, textBox.value);
+    createMarkerForDisplay(lat, lng, textBox.value);
+    editMarker.setMap(null);
```

Comment:
Good to clean up unused resources.
```suggestion
    editMarker.setMap(null);
    editMarker = null;
```

Reply:
Updated